${var@a} attribute transform errors on list/map values

The bash `${var@a}` parameter transformation (return the variable's attribute string) works on scalars but errors when the variable is a list or map.

## Repro (mode bash)
```
declare -A m=([x]=1)
echo "attr=${m@a}"
```
- **bash 5.3.9**: `attr=A`
- **zsh**: `bad substitution` (no `@a` support)
- **lush**: `error[E1133] type mismatch: scalar operator '@' applied to 'm', which is a map`

## Detail
Scalar attribute queries already work:
```
declare -i n=5  ; ${n@a}  -> i   (lush: correct)
declare -r ro=x ; ${ro@a} -> r   (lush: correct)
declare -A arr  ; ${arr@a} -> A   (lush: errors)
```
The `@` transform dispatch applies a scalar-kind guard before the attribute query. `@a` is an attribute query and is kind-agnostic: it must return the declared-attribute letters (A associative, a indexed, plus i/r/x/u/l/... as applicable) regardless of whether the value is scalar, list, or map.

## Scope
The `@`-transform handler in the expansion engine; route `@a` (and bash's related `@A` declare-reproduction form) ahead of the scalar-kind guard. The same fix verifies element-wise transforms on vectors (`${arr[@]@U}` etc.) behave correctly.

Found by the differential harness, seed `bash/217_bash_param_transform.sh`.

--- a/src/executor.c
+++ b/src/executor.c
@@ -13832,7 +13832,17 @@ static char *parse_parameter_expansion(executor_t *executor,
         /// single-element. The bare-name check fires only when the
         /// operator's left operand is a complete collection
         /// identifier with no subscript.
-        if (var_len > 0) {
+        ///
+        /// Exception: the @a attribute query is metadata-only and
+        /// kind-agnostic. It reports the declared-attribute letters
+        /// (a / A for indexed / associative arrays, plus i/r/x/...)
+        /// and must work on a collection, unlike the value-shaped @
+        /// transforms (@Q/@E/@U/...) which do need an explicit [@] to
+        /// vectorize. Let @a fall through to the case 17 dispatch,
+        /// where get_variable_attributes handles every kind even when
+        /// the scalar value lookup misses (arrays have a NULL scalar).
+        bool attr_query = (op_type == 17 && op_pos[1] == 'a');
+        if (var_len > 0 && !attr_query) {
             array_value_t *bare_array = symtable_get_array(var_name);
             if (bare_array) {
                 const char *op_str = operators[op_type];

--- a/tests/unit/test_executor.c
+++ b/tests/unit/test_executor.c
@@ -3077,6 +3077,38 @@ TEST(rt_pe_catalog_at_transform_on_bare_list) {
     ASSERT_EXIT_STATUS(r, 1);
 }
 
+TEST(rt_pe_at_attr_query_on_map) {
+    /// ${map@a} is the kind-agnostic attribute query: it reports the
+    /// declared-attribute letters and must work on a collection, unlike
+    /// the value-shaped @ transforms above which require [@] (#121).
+    run_result_t r = run_shell("declare -A m=([x]=1)\n"
+                               "echo \"[${m@a}]\"\n");
+    ASSERT_STDOUT_EQ(r, "[A]\n");
+}
+
+TEST(rt_pe_at_attr_query_on_list) {
+    run_result_t r = run_shell("declare -a idx=(a b c)\n"
+                               "echo \"[${idx@a}]\"\n");
+    ASSERT_STDOUT_EQ(r, "[a]\n");
+}
+
+TEST(rt_pe_at_attr_query_scalar_unaffected) {
+    /// The bare-name guard exception is limited to @a; the scalar
+    /// attribute paths are unchanged.
+    run_result_t r = run_shell("declare -i n=5\n"
+                               "declare -r ro=locked\n"
+                               "echo \"[${n@a}][${ro@a}]\"\n");
+    ASSERT_STDOUT_EQ(r, "[i][r]\n");
+}
+
+TEST(rt_pe_at_value_transform_on_map_still_errors) {
+    /// Only @a bypasses the guard; a value-shaped @ transform on a bare
+    /// collection remains a type error (needs [@] to vectorize).
+    run_result_t r = run_shell("declare -A m=([x]=1)\n"
+                               "echo \"[${m@U}]\"\n");
+    ASSERT_EXIT_STATUS(r, 1);
+}
+
 TEST(rt_pe_catalog_conditional_on_map) {
     run_result_t r = run_shell("declare -A m=([a]=1 [b]=2)\n"
                                "echo \"[${m:-default}]\"\n");
@@ -4092,6 +4124,10 @@ int main(void) {
     RUN_TEST(rt_pe_catalog_pattern_strip_on_bare_list);
     RUN_TEST(rt_pe_catalog_substring_on_bare_list);
     RUN_TEST(rt_pe_catalog_at_transform_on_bare_list);
+    RUN_TEST(rt_pe_at_attr_query_on_map);
+    RUN_TEST(rt_pe_at_attr_query_on_list);
+    RUN_TEST(rt_pe_at_attr_query_scalar_unaffected);
+    RUN_TEST(rt_pe_at_value_transform_on_map_still_errors);
     RUN_TEST(rt_pe_catalog_conditional_on_map);
     RUN_TEST(rt_pe_catalog_scalar_slice_still_works);
     RUN_TEST(rt_pe_catalog_per_element_case_mod_via_slice);

